app.rb

```diff
@@ -1,19 +1,68 @@
+require 'json'
 require_relative 'classroom'
 require_relative 'student'
 require_relative 'teacher'
 require_relative 'book'
 require_relative 'rentals'
 require_relative 'modules'
 require_relative 'create_teacher_student'
+require_relative 'data'
 
 class App
   include LibraryUtils
+  include SavingData
+  attr_reader :books, :people, :rentals
+
   def initialize
     @books = []
     @people = []
     @rentals = []
   end
 
+  def load_book_data(books_json)
+    books_json.each do |book|
+      @books << Book.new(book['title'], book['author'])
+    end
+  end
+
+  def load_people_data(people_json)
+    people_json.each do |person|
+      @people << if person['type'] == 'Teacher'
+                   Teacher.new(person['age'], person['specialization'], person['name'])
+                 elsif person['type'] == 'Student'
+                   parent_info = person['parent_permission'] || true
+                   Student.new(person['age'], 'first', person['name'], parent_permission: parent_info)
+                 end
+    end
+  end
+
+  def load_rental_data(rentals_json)
+    rentals_json.each do |rent|
+      rentee = @people.select { |person| person.name == rent['person_name'] }
+      rented_book = @books.select { |book| book.title == rent['book_title'] }
+      @rentals << Rental.new(rent['date'], rented_book[0], rentee[0])
+    end
+  end
+
+  def load_file_data
+    books_json = JSON.parse(fetch_file('books'))
+    people_json = JSON.parse(fetch_file('people'))
+    rentals_json = JSON.parse(fetch_file('rentals'))
+    load_book_data(books_json)
+    load_people_data(people_json)
+    load_rental_data(rentals_json)
+  end
+
+  def fetch_file(filename)
+    if File.exist?("jsons/#{filename}.json")
+      File.read("jsons/#{filename}.json")
+    else
+      json_file = [].to_json
+      File.write("jsons/#{filename}.json", json_file)
+      json_file
+    end
```

Comment:
- [ ] I got this error message after I deleted `books.json` file and tried to start up the app. Perhaps you need to cater for a situation whereby the database is empty. 👍 

<img width="578" alt="Untitled" src="https://github.com/adeola003/OOP-School-library/assets/103386183/e1111da2-2ab3-4e3d-9c87-e81b4a36a7ee">

Reply:
We weren't asked this in the project requirements

Reply:
Hi @adeola003 it is a requirement that you cater to the situation where the files do not exist.

![image](https://github.com/adeola003/OOP-School-library/assets/81468675/16c42706-513a-44e8-a10b-284350db0137)